Should fake time work outside main thread if specified explicitly?

Hi,

While debugging an [issue on a different project](https://github.com/pytest-dev/pytest-xdist/issues/56) I used `python-libfaketime` and accidentally noticed the `fake_time` context manager [has an argument](https://github.com/simon-weber/python-libfaketime/blob/master/libfaketime/__init__.py#L104) `only_main_thread=True`. Curious I tried `only_main_thread=False`, but got an exception:

```
self = <libfaketime.fake_time object at 0x107308c50>

    def _should_patch_uuid(self):
        return hasattr(uuid, '_uuid_generate_time') and \
>               not self._prev_spec and \
                self._should_fake()
E       AttributeError: 'fake_time' object has no attribute '_prev_spec'
```

At first I thought it is just simply the case of `_prev_spec` missing and changed it to `getattr(self, '_prev_spec', None)`, but now, upon closer inspection of the issue, it is [only being set](https://github.com/simon-weber/python-libfaketime/blob/master/libfaketime/__init__.py#L114) when `only_main_thread=True`. 

Therefore before submitting any pull request I wonder - is this expected behavior?

--- a/libfaketime/__init__.py
+++ b/libfaketime/__init__.py
@@ -111,12 +111,12 @@ class fake_time(ContextDecorator):
         self.time_to_freeze = _datetime  # freezegun compatibility
 
     def _should_fake(self):
-        return self.only_main_thread and threading.current_thread().name == 'MainThread'
+        return not self.only_main_thread or threading.current_thread().name == 'MainThread'
 
     def _should_patch_uuid(self):
         return hasattr(uuid, '_uuid_generate_time') and \
-                not self._prev_spec and \
-                self._should_fake()
+               self._should_fake() and \
+               not self._prev_spec
 
     def _format_datetime(self, _datetime):
         return _datetime.strftime('%Y-%m-%d %T %f')

